Download button crash in library

### Steps to reproduce

java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:558)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:942)
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	... 1 more
Caused by: java.nio.file.NoSuchFileException: /storage/emulated/0/Android/data/com.github.libretube/files/audio/Journey - Don't Stop Believin' (Official Audio)_126_kbps_M4A.mp4
	at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
	at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:144)
	at sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:115)
	at j$.nio.file.spi.a.x(Unknown Source:14)
	at j$.nio.file.Files.readAttributes(SourceFile:0)
	at j$.nio.file.Files.size(Unknown Source:5)
	at com.github.libretube.ui.adapters.DownloadsAdapter.onBindViewHolder(DownloadsAdapter.kt:101)
	at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:1157)
	at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:59)
	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1)
	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:54)
	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:825)
	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:79)
	at androidx.recyclerview.widget.RecyclerView.onMeasure(RecyclerView.java:64)
	at android.view.View.measure(View.java:26368)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
	at android.view.View.measure(View.java:26368)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
	at android.view.View.measure(View.java:26368)
	at androidx.constraintlayout.widget.ConstraintLayout$Measurer.measure(ConstraintLayout.java:547)
	at androidx.constraintlayout.core.widgets.ConstraintWidgetContainer.measure(ConstraintWidgetContainer.java:253)
	at androidx.constraintlayout.core.widgets.analyzer.Direct.verticalSolvingPass(Direct.java:31)
	at androidx.constraintlayout.core.widgets.analyzer.Direct.solveVerticalMatchConstraint(Direct.java:102)
	at androidx.constraintlayout.core.widgets.analyzer.Direct.verticalSolvingPass(Direct.java:186)
	at androidx.constraintlayout.core.widgets.analyzer.Direct.verticalSolvingPass(Direct.java:460)
	at androidx.constraintlayout.core.widgets.ConstraintWidgetContainer.layout(ConstraintWidgetContainer.java:440)
	at androidx.constraintlayout.core.widgets.analyzer.BasicMeasure.solveLinearSystem(BasicMeasure.java:34)
	at androidx.constraintlayout.widget.ConstraintLayout.resolveSystem(ConstraintLayout.java:1433)
	at androidx.constraintlayout.widget.ConstraintLayout.onMeasure(ConstraintLayout.java:509)
	at androidx.constraintlayout.motion.widget.MotionLayout.onMeasure(MotionLayout.java:90)
	at android.view.View.measure(View.java:26368)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
	at androidx.appcompat.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:160)
	at android.view.View.measure(View.java:26368)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1608)
	at android.widget.LinearLayout.measureVertical(LinearLayout.java:878)
	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
	at android.view.View.measure(View.java:26368)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
	at android.view.View.measure(View.java:26368)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1608)
	at android.widget.LinearLayout.measureVertical(LinearLayout.java:878)
	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
	at android.view.View.measure(View.java:26368)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
	at com.android.internal.policy.DecorView.onMeasure(DecorView.java:760)
	at android.view.View.measure(View.java:26368)
	at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:3929)
	at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:2615)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2887)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2331)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9095)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1319)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1327)
	at android.view.Choreographer.doCallbacks(Choreographer.java:985)
	at android.view.Choreographer.doFrame(Choreographer.java:916)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1302)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7918)
    

### Expected behavior

Should show what I downloaded

### Actual behavior

Instead libretube app shutdown

### LibreTube version

0.13.0

### Android version

Android 13

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

--- a/app/src/main/java/com/github/libretube/services/OfflinePlayerService.kt
+++ b/app/src/main/java/com/github/libretube/services/OfflinePlayerService.kt
@@ -23,6 +23,7 @@ import com.github.libretube.util.NowPlayingNotification
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
+import kotlin.io.path.exists
 
 /**
  * A service to play downloaded audio in the background
@@ -85,9 +86,11 @@ class OfflinePlayerService : LifecycleService() {
             .build()
             .loadPlaybackParams(isBackgroundMode = true)
 
-        val audioItem = downloadWithItem.downloadItems.firstOrNull { it.type == FileType.AUDIO }
+        val audioItem = downloadWithItem.downloadItems.filter { it.path.exists() }
+            .firstOrNull { it.type == FileType.AUDIO }
             ?: // in some rare cases, video files can contain audio
-            downloadWithItem.downloadItems.firstOrNull { it.type == FileType.VIDEO } ?: return false
+            downloadWithItem.downloadItems.firstOrNull { it.type == FileType.VIDEO }
+            ?: return false
 
         val mediaItem = MediaItem.Builder()
             .setUri(audioItem.path.toAndroidUri())

--- a/app/src/main/java/com/github/libretube/ui/activities/OfflinePlayerActivity.kt
+++ b/app/src/main/java/com/github/libretube/ui/activities/OfflinePlayerActivity.kt
@@ -40,6 +40,7 @@ import com.github.libretube.util.OfflineTimeFrameReceiver
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
+import kotlin.io.path.exists
 
 @androidx.annotation.OptIn(androidx.media3.common.util.UnstableApi::class)
 class OfflinePlayerActivity : BaseActivity() {
@@ -133,7 +134,7 @@ class OfflinePlayerActivity : BaseActivity() {
             val downloadInfo = withContext(Dispatchers.IO) {
                 Database.downloadDao().findById(videoId)
             }
-            val downloadFiles = downloadInfo.downloadItems
+            val downloadFiles = downloadInfo.downloadItems.filter { it.path.exists() }
             playerBinding.exoTitle.text = downloadInfo.download.title
             playerBinding.exoTitle.isVisible = true
 

--- a/app/src/main/java/com/github/libretube/ui/adapters/DownloadsAdapter.kt
+++ b/app/src/main/java/com/github/libretube/ui/adapters/DownloadsAdapter.kt
@@ -25,6 +25,7 @@ import kotlin.io.path.deleteIfExists
 import kotlin.io.path.fileSize
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.runBlocking
+import kotlin.io.path.exists
 
 class DownloadsAdapter(
     private val context: Context,
@@ -50,7 +51,7 @@ class DownloadsAdapter(
             videoInfo.text = download.uploadDate?.let { TextUtils.localizeDate(it) }
 
             val downloadSize = items.sumOf { it.downloadSize }
-            val currentSize = items.sumOf { it.path.fileSize() }
+            val currentSize = items.filter { it.path.exists() }.sumOf { it.path.fileSize() }
 
             if (downloadSize == -1L) {
                 progressBar.isIndeterminate = true

